.github/workflows/reusable-build.yml

```diff
@@ -34,4 +34,6 @@ jobs:
       - name: Run unit tests
         run: make test
       - name: Upload coverage to codecov.io
-        uses: codecov/codecov-action@v4
\ No newline at end of file
+        uses: codecov/codecov-action@v4
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
what's the purpose of this PR? Is there a related issue regarding this?

Reply:
Added issue link in the PR discription.

Reply:
hmm, according to the doc: https://github.com/codecov/codecov-action?tab=readme-ov-file#breaking-changes:
there's a breaking change on the action, but they also stated that `PRs made from forks to the upstream public repos will support tokenless`. That means we don't need the token?

Reply:
Tokenless is only supported in PRs not on the `main` branch